pills/04-basics-of-language.xml

```diff
@@ -4,5 +4,362 @@
       version="5.0"
       xml:id="basics-of-language">
 
-<title>basics of language</title>
+  <title>The basics of language</title>
+
+  <para>
+    Welcome to the fourth Nix pill. In the previous
+    <link linkend="enter-environment">third pill</link> we entered the Nix
+    environment. We installed software as user, managed the profile, switched
+    between generations, and queried the nix store. That's the very basics of
+    nix administration somehow.
+  </para>
+
+  <para>
+    The
+    <link xlink:href="http://nixos.org/nix/manual/#chap-writing-nix-expressions">Nix language</link>
+    is used to write derivations. The
+    <link xlink:href="http://nixos.org/nix/manual/#sec-nix-build">nix-build</link>
+    tool is used to build derivations. Even as a system administrator that
+    wants to customize the installation, it's necessary to master Nix. Using
+    Nix for your jobs means you get the features we saw in the previous pills
+    for free.
+  </para>
+
+  <para>
+    The syntax is very uncommon thus looking at existing examples may lead to
+    thinking that there's a lot of magic behind. In reality, it's only about
+    writing utility functions for making things convenient.
+  </para>
+
+  <para>
+    On the other hand, this same syntax is great for describing packages.
+  </para>
+
+  <para>
+    <emphasis role="underline">Important</emphasis>: in Nix, everything is an
+    expression, there are no statements. This is common to many functional
+    languages.
+  </para>
+
+  <para>
+    <emphasis role="underline">Important</emphasis>: values in Nix are
+    immutable.
+  </para>
+
+  <section>
+    <title>Value types</title>
+
+    <para>
+      We've installed nix-repl in the previous pill. If you didn't,
+      <literal>nix-env -i nix-repl</literal>. The nix-repl syntax is slightly
+      different than nix syntax when it comes to assigning variables, but no
+      worries.  I prefer playing with nix-repl with you before cluttering your
+      mind with more complex expressions.
+    </para>
+
+    <para>
+      Launch nix-repl. First of all, nix supports basic arithmetic operations:
+      +, -, and *. The integer division can be done with builtins.div.
```

Comment:
I wonder if these symbols would be better as `<token>` or `<function>`, same as other operators below.